syntax = "proto3";

package task;

service TaskService {
  rpc CreateTask(CreateTaskRequest) returns (TaskResponse);
  rpc GetTask(GetTaskRequest) returns (TaskResponse);
  rpc UpdateTask(UpdateTaskRequest) returns (TaskResponse);
  rpc DeleteTask(DeleteTaskRequest) returns (Empty);
  rpc ListTasks(Empty) returns (ListTasksResponse);
}

message Task {
  string id = 1;
  string title = 2;
  string description = 3;
  string status = 4; // e.g., "pending", "completed"
}

message CreateTaskRequest {
  string title = 1;
  string description = 2;
}

message UpdateTaskRequest {
  string id = 1;
  string title = 2;
  string description = 3;
  string status = 4;
}

message GetTaskRequest {
  string id = 1;
}

message TaskResponse {
  Task task = 1;
}

message DeleteTaskRequest {
  string id = 1;
}

message ListTasksResponse {
  repeated Task tasks = 1;
}

message Empty {}

